### PR TITLE
SOLR-15871-Update log4j to 2.17 to address CVE-2021-45105

### PR DESCRIPTION
**Pull request by weidongkl**
https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-45105

Signed-off-by: weidong <weidong@uniontech.com>

https://issues.apache.org/jira/browse/SOLR-15871

# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -488,7 +488,7 @@ Bug Fixes
 
 * SOLR-8319: Fix NPE in pivot facets, add non-Analyzed query method in FieldType. (Houston Putman, Isabelle Giguere)
 
-* SOLR-15843: Update Log4J to 2.16 (Mike Drob, janhoy)
+* SOLR-15843: Update Log4J to 2.17 (Mike Drob, janhoy, weidongkl)
```
Comment (janhoy) [inline solr/CHANGES.txt:491]: 8.11.1 is released, so you need to file a new JIRA and place a changes entry in the 8.11.2 and 9.0 sections.

**Review by janhoy — CHANGES_REQUESTED** [review]

**Comment by janhoy** [comment]
@weidongkl are you able to do the changes, so this can be merged?

**Comment by weidongkl** [comment]
> @weidongkl are you able to do the changes, so this can be merged?

I'll do it as soon as possible

**Comment by janhoy** [comment]
Pushed outside of github

**Pull request closed without merging**